src/Microsoft.AspNetCore.Rewrite/Internal/PatternSegments/RewriteMapSegment.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System;
+
+using Microsoft.AspNetCore.Rewrite.Internal.IISUrlRewrite;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.Rewrite.Internal.PatternSegments
+{
+    public class RewriteMapSegment : PatternSegment
+    {
+        private readonly IISRewriteMap _rewriteMap;
+        private readonly Pattern _pattern;
+
+        public RewriteMapSegment(IISRewriteMap rewriteMap, Pattern pattern)
+        {
+            _rewriteMap = rewriteMap;
+            _pattern = pattern;
+        }
+
+        public override string Evaluate(RewriteContext context, MatchResults ruleMatch, MatchResults condMatch)
+        {
+            string key = _pattern.Evaluate(context, ruleMatch, condMatch);
+            string value;
+            if (_rewriteMap.TryGetEntry(key, out value))
```

Comment:
I think this is wrong. If we don't find an entry in our map, we shouldn't be throwing, it just means it wasn't found in our server variables. Ex I have a url: /foo/bar which maps to foo?bar=1. If our url doesn't equal /foo/bar, it's not an invalid url, it just means it wasn't found in our map.